src/uWS.cpp

```diff
@@ -18,10 +18,16 @@ using namespace std;
 
 #ifdef _WIN32
 #if _MSC_VER >= 1900
-#include <io.h>
-#if !defined(dup)
-#define dup _dup
-#endif
+#include <cassert>
+inline FD dup(FD socket)
+{
+ WSAPROTOCOL_INFO pi;
+ auto rv = WSADuplicateSocket(socket, ::GetCurrentProcessId(), &pi); 
+ assert(rv == 0);
+ auto socketDuped = WSASocket(AF_INET, SOCK_STREAM, IPPROTO_TCP, &pi, 0, WSA_FLAG_OVERLAPPED); 
+ assert(socketDuped > 0);
+ return socketDuped;  
+}
```

Comment:
This is a very nice addition, I'll take a look at this tomorrow. I'm super tired today.